src/bika/lims/browser/analysisrequest/add2.py

```diff
@@ -1557,6 +1558,40 @@ def ajax_recalculate_prices(self):
 
         return prices
 
+    def is_true(self, val):
+        """Returns whether val evaluates to True
+        """
+        val = str(val).strip().lower()
+        return val in ["y", "yes", "1", "true", "on"]
+
+    def is_date_field(self, field):
+        """Returns whether the field is for storing a date or datetime
+        """
+        return field.type in ["date", "datetime", "datetime_ng"]
+
+    def get_min_dt(self, field, default=None):
+        """Returns the minimum datetime supported for the given field
+        """
+        # find-out from the widget
+        widget = getattr(field, "widget", None)
+        no_past = getattr(widget, "datepicker_nopast", False)
+        if self.is_true(no_past):
+            return datetime.now()
+        return default
+
+    def get_max_dt(self, field, default=None):
+        """Returns the maximum datetime supported for the given field
+        """
+        # find-out from the widget
+        widget = getattr(field, "widget", None)
+        no_future = getattr(widget, "datepicker_nofuture", False)
+        if self.is_true(no_future):
+            return datetime.now()
+        two_months = getattr(widget, "datepicker_2months", False)
+        if self.is_true(two_months):
+            return datetime.now() + relativedelta(months=2)
```

Comment:
Where do we have this in place? Not sure if this is possible at all with the current native HTML 5 date widget.